src/main/java/HttpHandlerSelector.java

```diff
@@ -1,114 +1,24 @@
 package com.td.HttpServer;
-import java.util.*;
-import java.text.DateFormat;
-import java.text.SimpleDateFormat;
-import java.io.IOException;
 
-// TODO This class is too big, and has multiple responsibilities.
-//      It needs to be broken up into multiple classs.
-//      Two classes that could be pulled out of this are
-//         HttpResponseBuilder & HeaderBuilder
+import java.util.*;
 
 public class HttpHandlerSelector {
 
-  IFileIO fileIO;
-  HashMap<String, String> standardHeaders;
-
-  private final String CODE_200 = "HTTP/1.1 200 OK";
-  private final String CODE_404 = "HTTP/1.1 404 Not Found";
-  private final String CODE_400 = "HTTP/1.1 400 Bad Request";
-
-  private final String HTML = ".html";
+  HttpHandleRequest getRequest;
 
-  public HttpHandlerSelector(IFileIO fileIO) {
-    this.fileIO = fileIO;
-    standardHeaders = new HashMap<String, String>();
-    standardHeaders.put("Server", "Simple Http Server written in Java");
+  public HttpHandlerSelector(HttpHandleRequest getRequest) {
+    this.getRequest = getRequest;
   }
 
   public HttpResponse generateResponse(HttpRequest request) {
     if (request.method().equals("GET")) {
-      return getRequest(request);
-    }
-    return notFound(request);
-  }
-
-  public HttpResponse generate400Response() {
-    byte[] body = "400 : Bad Request".getBytes();
-    return new HttpResponse(CODE_400, body, buildHeaders(body, HTML));
-  }
-
-  private HttpResponse getRequest(HttpRequest request) {
-    String path = request.path();
-    if (path.equals("/")) {
-      return defaultResponse();
-    }
-    try {
-      byte[] body = fileIO.getContent(path);
-      HashMap<String, String> headers = buildHeaders(body, path);
-      return new HttpResponse(CODE_200, body, headers);
-    }
-    catch (IOException e) {
-      return notFound(request);
+      return getRequest.generateResponse(request);
```

Comment:
`getRequest` is a little confusing here—it makes me feel like you're calling this on a Request object that happens to be of the GET variety. Probably as simple as renaming to something like `getRequestHandler`